src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryTryGetValueFixer.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class CSharpPreferDictionaryTryGetValueFixer : PreferDictionaryTryGetValueFixer
+    {
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.FirstOrDefault();
+            var dictionaryAccessLocation = diagnostic?.AdditionalLocations[0];
+            if (dictionaryAccessLocation is null)
+            {
+                return;
+            }
+
+            Document document = context.Document;
+            SyntaxNode root = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+            var dictionaryAccessNode = root.FindNode(dictionaryAccessLocation.SourceSpan);
+            var dictionaryAccess = dictionaryAccessNode as ElementAccessExpressionSyntax ?? (dictionaryAccessNode as ArgumentSyntax)?.Expression as ElementAccessExpressionSyntax;
```

Comment:
You may not actually need this if you used `getInnermostNodeForTie`.

Reply:
Nice hint, thanks! Done.